### ISSUE TITLE
Weight normalization for tf versions that are incompatible with tfa

### ISSUE THREAD

**Issue by JuliaWasala** [krasserm/super-resolution#97]
Hi Martin,

Thanks for making the implementations of these models available! I have a question, I have to use tf 2.5.2 in order to be able to use gpu, and today I discovered that tensorflow-addons are not compatible with tf <2.6 (a bit of a mystery: I did a test run yesterday and it ran without errors).
Now , I've found your blogpost and decided to use the Conv2DWeightNorm from this repo (https://github.com/ychfan/tf_estimator_barebone/blob/master/common/layers.py) in order to  be able to run your code with as little adaptations as possible. I've also taken a look at your other branches, but if I'm correct those are tf1.
I'm having a few issues, though, and I was wondering whether you'd have an idea what the problem would be. For now I just pasted the code of Conv2DWeightNorm and conv2d_weight_norm in the wdsr.py file, changed tf.layers to tf.keras.layers, removed the arguments _scope and _reuse that don't exist in tf2, changed the relevant wdsr functions so they would call conv2d_weight_norm(input, **args) instead of conv2d_weightnorm(**args)(input). 
It does run, but I get an error that Gradients do not exist for variabes associated with the conv2d_weight_norm layers. 

would you be able to offer me advice on how to use the custom weightnorm layer with the code you currently have in master?

**Comment by krasserm** [comment]
> and today I discovered that tensorflow-addons are not compatible with tf <2.6

There are tensorflow-addons versions that are compatible if I understand correctly, see https://github.com/tensorflow/addons#python-op-compatibility-matrix for details

**Comment by JuliaWasala** [comment]
oof, that's great! I got thrown off by the error message of tf that seemed to say that tfa wasn't compatible with lower versions of tf at all! thanks a lot
